lib/msf/core/data_store.rb

```diff
@@ -169,29 +169,29 @@ def to_s(delim = ' ')
   def to_h
     datastore_hash = {}
     self.keys.each do |k|
-      datastore_hash[k.to_s] = self[k].to_s
+      datastore_hash[k.to_s.dup.force_encoding('UTF-8')] = self[k].to_s.dup.force_encoding('UTF-8')
     end
     datastore_hash
   end
 
   # Hack on a hack for the external modules
-  def to_nested_values
+  def to_external_message_h
     datastore_hash = {}
 
     array_nester = ->(arr) do
       if arr.first.is_a? Array
         arr.map &array_nester
       else
-        arr.map &:to_s
+        arr.map { |item| item.to_s.dup.force_encoding('UTF-8') } # &:to_s
```

Comment:
Looks like debugging code is left in here 👀 

```suggestion
        arr.map { |item| item.to_s.dup.force_encoding('UTF-8') }
```

Reply:
Just confirming if you were wanting to convert `nil` into an empty string here

Reply:
As discussed; Let's leave the nil -> empty string conversion here as it might introduce regressions :+1: